server/dataone_package.py

```diff
@@ -24,57 +29,95 @@ def create_resource_map(resmap_pid, eml_pid, file_pids):
     :rtype: bytes
     """
 
-    print("FILE_PIDS      ", file_pids)
     logger.debug('Entered create_resource_map')
+#    logger.debug('resmap_pids: {}'.format(resmap_pid))
+#    logger.debug('eml_pids: {}'.format(eml_pid))
+#    logger.debug('file_pids {}'.format(file_pids))
     res_map = createSimpleResourceMap(resmap_pid, eml_pid, file_pids)
     # createSimpleResourceMap returns type d1_common.resource_map.ResourceMap
 
     logger.debug('Leaving create_resource_map')
     return res_map.serialize()
 
 
-def create_minimum_eml(tale, user):
+def create_minimum_eml(tale, user, item_ids, eml_pid):
     """
-    Creates a bare minimum EML record for a package. This includes the title,
-    creator, and contact.
+    Creates a bare minimum EML record for a package.
 
     :param tale: The tale that is being packaged.
     :param user: The user that hit the endpoint
+    :param item_ids: A lsit of the item ids of the objects that are going to be packaged
+    :param eml_pid: The PID for the eml document. Assume that this is the package doi
     :type tale: wholetale.models.tale
     :type user: girder.models.user
+    :type item_ids: list
+    :type eml_pid: str
     :return: The EML as as string of bytes
     :rtype: bytes
     """
 
-    logger.debug('Entered create_minimum_eml')
-    top = '<?xml version="1.0" encoding="UTF-8"?>'
-    namespace = '<eml:eml xmlns:eml="eml://ecoinformatics.org/eml-2.1.1" ' \
-                'xmlns:stmml="http://www.xml-cml.org/schema/stmml-1.1" xmlns:' \
-                'xsi="http://www.w3.org/2001/XMLSchema-instance" packageId="' \
-                'doi:10.18739/A20Q25" system="https://arcticdata.io" xsi:' \
-                'schemaLocation="eml://ecoinformatics.org/eml-2.1.1 eml.xsd">'
-
-    dataset = '<dataset>\n'
-    title = '<title>{0}</title>\n'.format(str(tale.get('title', '')))
-
-    individual_name = '<individualName>\n<surName>\n{0}\n</surName>\n</individualName>'.format(
-        str(user.get('lastName', '')))
-
-    creator = '<creator>\n{0}\n</creator>\n'.format(individual_name)
-    contact = '<contact>\n{0}\n</contact>\n'.format(individual_name)
-    dataset_close = '</dataset>\n'
-    eml_close = '</eml:eml>'
-
-    # Append the above xml together to form the EML document
-    xml = top+namespace+dataset+title+creator+contact+dataset_close+eml_close
-
+    # The surName in the EML record is taken by the user's last name. If
+    # it doesn't exist, try to use the first name.
+    lastName = user.get('lastName', 'None')
+    firstName = user.get('firstName', 'None')
+    ns = ET.Element('eml:eml')
+
+    ns.set('xmlns:eml', "eml://ecoinformatics.org/eml-2.1.1")
+    ns.set('xsi:schemaLocation', "eml://ecoinformatics.org/eml-2.1.1 eml.xsd")
+    ns.set('xmlns:stmml', "http://www.xml-cml.org/schema/stmml-1.1")
+    ns.set('xmlns:xsi', "http://www.w3.org/2001/XMLSchema-instance")
+    ns.set('scope', "system")
+    ns.set('system', "knb")
```

Comment:
For testing and development, 'knb' is fine but ideally this changes depending on which MN you're creating Objects on. The MN's `nodeId` is a good option, with the `urn:node:` removed.

Reply:
This value will have to eventually be grabbed from the publish dialog in the UI when the user chooses which member node to publish to. Until then I was planning on using `knb`

Reply:
Cool, sounds good.